core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -28,6 +28,64 @@ oppia.constant(
 oppia.constant(
   'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE', '/explorationsummarieshandler/data');
 
+// Sorts the node list for display in collection editor frontend. The list
+// starts with a startNode which contains no prerequisite skills. After that
+// nodes are added to the list if their prerequisite skills match the acquired
+// skills of the last node that was added to the node list.
+oppia.filter('customSorter', function() {
+  return function(nodes) {
+    // Get starting node
+    var startingNode;
+    for (var i = 0; i < nodes.length; i++) {
+      if (nodes[i].getPrerequisiteSkillList().isEmpty()) {
+        startingNode = nodes[i];
+        break;
+      }
+    }
+
+    // Populate node list
+    var nodeList = [];
+    nodeList.push(startingNode);
+    while (true) {
+      for (var i = 0; i < nodes.length; i++) {
+        var nodePrerequisiteSkills =
+          nodes[i].getPrerequisiteSkillList()._skillList;
+        var currNodeAcquiredSkills =
+          nodeList[nodeList.length - 1].getAcquiredSkillList()._skillList;
+        if (!nodePrerequisiteSkills.length == 0 && compareTwoArrays(
```

Comment:
Instead of using ``compareTwoArrays`` at all, could you maybe use SkillList's [isSupersetOfSkillList()](https://github.com/amgowano/oppia/blob/1ecb7f7198a3ae8705199a486ddbe7dc53cfba69/core/templates/dev/head/domain/collection/SkillListObjectFactory.js#L144)?

Also, did you mean to do ``!nodePrerequisiteSkills.length``?